### PR TITLE
add burn function

### PR DESCRIPTION
**Pull request by askmike**

### CODE REVIEW

File: contracts/Ico.sol [hunk 1]
```diff
@@ -85,14 +88,24 @@ contract Ico is BasicToken {
...
+    tokensIssued = tokensIssued.sub(_value);
+    Burn(msg.sender, _value);
     return true;
   }
 
   /**
    * Withdraw all funds and kill fund smart contract
    */
-  function liquidate() onlyOwner returns (bool) {
+  function liquidate() public onlyOwner returns (bool) {
```
Comment (ale316) [inline contracts/Ico.sol]: We've decided the contract will not keep any reserve right? aka the liquidate function actually only needs to self-destruct?

File: contracts/Ico.sol [hunk 2]
```diff
@@ -85,14 +88,24 @@ contract Ico is BasicToken {
     tokensFrozen = tokensIssued * 2;
 
     owner.transfer(ethAmount);
+    Participate(msg.sender, numTokens);
+    return true;
+  }
+
+  function burn(uint256 _value) public returns (bool) {
```
Comment (ale316) [inline contracts/Ico.sol]: maybe we should rename `_value` into `_amount`? no biggie though

File: contracts/Ico.sol [hunk 3]
```diff
@@ -85,14 +88,24 @@ contract Ico is BasicToken {
...
+    require(_value <= balances[msg.sender]);
 
+    // SafeMath.sub will throw if there is not enough balance.
+    balances[msg.sender] = balances[msg.sender].sub(_value);
+    tokensIssued = tokensIssued.sub(_value);
```
Comment (ale316) [inline contracts/Ico.sol]: `aum` also needs to be adjusted I think, as when we burn we are also sending ether to the addresses that are claiming it (offchain)

File: contracts/Ico.sol (hunk 2 above)
Comment (askmike) [reply contracts/Ico.sol]: agree, will do.

File: contracts/Ico.sol (hunk 3 above)
Comment (askmike) [reply contracts/Ico.sol]: I'll merge master into this and add it :)

File: contracts/Ico.sol (hunk 1 above)
Comment (askmike) [reply contracts/Ico.sol]: People can still send attach ether to other public functions (like transfer, balanceOf). If they fuck up we should be able to get the ether back.

**Pull request merged**